### PR TITLE
Release all resources used in tests

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
- Make sure all resources are released (**ALL** changes are in **Unit Tests**)
- Fix null dereference by updating
```
private Optional<InputStream> nonCloseableInputStream(Optional<InputStream> toBeClosed) {
        return toBeClosed.map(is -> ReleasableInputStream.wrap(is).disableClose());
     } 
```
to return optional 

## Testing
`mvn clean install`

### CODE REVIEW

File: core/src/test/java/software/amazon/awssdk/core/auth/EC2MetadataServiceMock.java [hunk 1]
```diff
@@ -171,6 +173,7 @@ public void run() {
                 } finally {
                     try {
                         outputStream.close();
+                        socket.close();
```
Comment (dagnir) [inline core/src/test/java/software/amazon/awssdk/core/auth/EC2MetadataServiceMock.java]: If the first close throws this won't get cleaned up

File: services/s3/src/test/java/software/amazon/awssdk/services/s3/handlers/PutObjectHeaderTest.java [hunk 2]
```diff
@@ -77,25 +78,31 @@ public void putObjectBytes_headerShouldContainContentType() {
     @Test
     public void putObjectFile_headerShouldContainContentType() throws IOException {
         File file = new RandomTempFile("test.html", 10);
```
Comment (varunnvs92) [inline services/s3/src/test/java/software/amazon/awssdk/services/s3/handlers/PutObjectHeaderTest.java]: Random file calls "deleteOnExit()" internally. I am okay with explicitly deleting the file. I am curious if fortify mentioned this change?

Comment (zoewangg) [reply services/s3/src/test/java/software/amazon/awssdk/services/s3/handlers/PutObjectHeaderTest.java]: No, it didn't mention it. I caught it and didn't know it got deleted automatically.  I will remove it in this case.

File: core/src/test/java/software/amazon/awssdk/core/auth/EC2MetadataServiceMock.java (hunk 1 above)
Comment (zoewangg) [reply core/src/test/java/software/amazon/awssdk/core/auth/EC2MetadataServiceMock.java]: good point, fixed!

**Pull request merged**